spec/models/spree/calculator/flexi_rate_spec.rb

```diff
@@ -1,13 +1,18 @@
 require 'spec_helper'
 
 describe Spree::Calculator::FlexiRate do
-  let(:calculator) { Spree::Calculator::FlexiRate.new }
+  let(:calculator_first) { Spree::Calculator::FlexiRate.new(preferred_first_item: 2, preferred_additional_item: 1, preferred_max_items: 3.0) }
+  let(:calculator_second) { Spree::Calculator::FlexiRate.new(preferred_first_item: 2, preferred_additional_item: 1, preferred_max_items: 5.0) }
+
   let(:line_item) { instance_double(Spree::LineItem, amount: 10, quantity: 4) }
 
   describe "computing for a single line item" do
-    it "returns the first item rate" do
-      calculator.stub preferred_first_item: 1.0
-      calculator.compute(line_item).round(2).should == 1.0
+    it "returns the first item rate when quantity is above max" do
+      expect(calculator_first.compute(line_item).round(2)).to eq(4.0)
+    end
+
+    it "returns the first item rate when quantity is below max" do
+      expect(calculator_second.compute(line_item).round(2)).to eq(5.0)
```

Comment:
Just for the sake of playing with rspec DSL I would suggest using `context`:
```ruby
describe Spree::Calculator::FlexiRate do
  let(:line_item) { instance_double(Spree::LineItem, amount: 10, quantity: 4) }
  let(:calculator) do
    Spree::Calculator::FlexiRate.new(
      preferred_first_item: 2,
      preferred_additional_item: 1,
      preferred_max_items: preferred_max_items
    )
  end

  context 'when the calculator has preferred max items set to 3.0' do
   let(:preferred_max_items) { 3.0 }

   it "returns the first item rate when quantity is above max" do
     expect(calculator.compute(line_item).round(2)).to eq(4.0)
   end
  end

  context 'when the calculator has preferred max items set to 4.0' do
   let(:preferred_max_items) { 4.0 }

   it "returns the first item rate when quantity is below max" do
     expect(calculator.compute(line_item).round(2)).to eq(5.0)
   end
  end
end
```
This will give you more "surface" to better describe what's different in each context. Using "first" and "second" doesn't tell much. Actually I would even use a more clear description for the `context` blocks compared to what I suggest, but you'll have a better understanding of the domain.